### DISCUSSION TITLE
Is there a way to display matching file information like file size, creation data etc

### DISCUSSION THREAD

**Discussion by ket000** [BurntSushi/ripgrep#2470] [General] [answered]
I can use -l to display filenames. However I am not sure if it can display file size , modified date etc. Basically running ls -l on returned file name that match the search. I can do it by passing to ls command, but just wanted to know if there is a built in support for this.

**Comment by BurntSushi** [comment, marked as answer]
No, there isn't. I don't think there ever will be. You'll want to use `ls` or `stat`. `man stat` for example reveals a very long list of file metadata information. It would be impractical and unwise for ripgrep to reinvent this.

**Comment by ket000** [comment]
Thanks for confirming this as I am already using ls to display the info I need. My use case was to display filenames that has specific string and then display additional items and filesize from these files. I was able to use -l to get filenames and then eventully grep and ls command to display what I need. Just wanted confirmation that I am not doing any additional steps that are not needed.

**Reply by BurntSushi** [reply]
Aye. And note you should be able to use `rg` instead of `grep` in that shell pipeline. :)

**Reply by ket000** [reply]
Yes, I can, but since I know already filename nd it is a vert simple search I have used grep. rg requires a folder name and a pattern to restrict search for a specific file. Since I already know the filename, I have used grep.

**Reply by BurntSushi** [reply]
ripgrep does not require a directory. It searches pain files too. And stdin. It works just like grep in those cases.
